js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.js

```diff
@@ -14,85 +18,79 @@ import AppButtonModalTrigger from '.~/components/app-button-modal-trigger';
 import groupShippingRatesByMethodFreeShippingThreshold from './groupShippingRatesByMethodFreeShippingThreshold';
+import {
+	getMinimumOrderAddHandler,
+	getMinimumOrderChangeHandler,
+	getMinimumOrderDeleteHandler,
+} from './handlers';
 import './minimum-order-card.scss';
 
 const MinimumOrderCard = ( props ) => {
-	const { value = [], onChange = noop } = props;
-	const nonZeroShippingRates = value.filter( isNonFreeFlatShippingRate );
-	const groups = groupShippingRatesByMethodFreeShippingThreshold(
-		nonZeroShippingRates
-	);
-	const countryOptions = nonZeroShippingRates.map(
-		( shippingRate ) => shippingRate.country
-	);
-
-	const handleEditChange = ( oldGroup ) => ( newGroup ) => {
-		const newValue = value.map( ( shippingRate ) => {
-			const newShippingRate = {
-				...shippingRate,
-				options: {
-					...shippingRate.options,
-				},
-			};
-
-			if ( newGroup.countries.includes( newShippingRate.country ) ) {
-				/**
-				 * Shipping rate's country exists in the new value countries,
-				 * so we just assign the new value threshold.
-				 */
-				newShippingRate.options.free_shipping_threshold =
-					newGroup.threshold;
-			} else if (
-				oldGroup.countries.includes( newShippingRate.country )
-			) {
-				/**
-				 * Shipping rate's country does not exist in the new value countries,
-				 * but it exists in the old value countries.
-				 * This means users removed the country in the edit modal,
-				 * so we set the threshold value to undefined.
-				 */
-				newShippingRate.options.free_shipping_threshold = undefined;
-			}
-
-			return newShippingRate;
-		} );
-
-		onChange( newValue );
-	};
-
-	const handleAddChange = ( newGroup ) => {
-		const newValue = value.map( ( shippingRate ) => {
-			const newShippingRate = {
-				...shippingRate,
-				options: {
-					...shippingRate.options,
-				},
-			};
-
-			if ( newGroup.countries.includes( newShippingRate.country ) ) {
-				newShippingRate.options.free_shipping_threshold =
-					newGroup.threshold;
-			}
-
-			return newShippingRate;
-		} );
-
-		onChange( newValue );
+	const { value = [], onChange } = props;
+
+	/**
+	 * Render an Edit button that displays EditMinimumOrderFormModal upon click.
+	 *
+	 * @param {Object} props Props.
+	 * @param {Array<CountryCode>} props.countryOptions Country options to be passed to EditMinimumOrderFormModal.
+	 * @param {MinimumOrderGroup} props.group Minimum order group to be edited.
+	 */
+	const renderGroupEditModalButton = ( { countryOptions, group } ) => {
+		return (
+			<AppButtonModalTrigger
+				button={
+					<AppButton isTertiary>
+						{ __( 'Edit', 'google-listings-and-ads' ) }
+					</AppButton>
+				}
+				modal={
+					<EditMinimumOrderFormModal
+						countryOptions={ countryOptions }
+						initialValues={ group }
+						onSubmit={ getMinimumOrderChangeHandler(
+							value,
+							onChange
+						)( group ) }
+						onDelete={ getMinimumOrderDeleteHandler(
+							value,
+							onChange
+						)( group ) }
```

Comment:
At this point, if we are going to provide all those arguments here, why not simply?:
```suggestion
						onSubmit={ getMinimumOrderChangeHandler(
							value,
							onChange,
							group ) }
						onDelete={ getMinimumOrderDeleteHandler(
							value,
							onChange,
							group ) }
```

Reply:
Thanks @tomalec , I've read your reasonings in https://github.com/woocommerce/google-listings-and-ads/pull/1410#discussion_r861957968, and I think you have some good and valid points there 🙂 👍  I'll make the changes like what you suggested here (i.e. `onSubmit={ getMinimumOrderChangeHandler( value, onChange, group ) }`).

Reply:
I have made the changes in [`5d03559` (#1410)](https://github.com/woocommerce/google-listings-and-ads/pull/1410/commits/5d0355925e568082f0d32a3d522429527b463f22) and [`c6f4f8f` (#1410)](https://github.com/woocommerce/google-listings-and-ads/pull/1410/commits/c6f4f8fd5bc63e507cb758fa2860eafa293eb7a0).